### PR TITLE
Fix privacy and add reusable CI

### LINKED ISSUE #79: Missing privacy api changes for custom cert changes.
copied from comment on #74 by @ewallah

When you run the core_privacy\privacy\provider_test::test_table_coverage test you see a warning:

The following tables with user fields must be covered with metadata providers:
  - local_recompletion_ccert_is (userid)

### PR DESCRIPTION
**Pull request by keevan**
- fix: privacy API issues for ccert_is table
- chore: add reusable CI workflows
- docs: add CI badge to README

Resolves #79

### CODE REVIEW

**Comment by keevan** [comment]
The CI badge would look like this on the README. Let me know if you want it tweaked / reordered.

image

**Comment by keevan** [comment]
Local tests before:

```
❯ ctrl test --filter 'provider_test::test_table_coverage'
Moodle 4.1.5+ (Build: 20230826), bababe5ab3dfac3528ab55728ca8ee9315a06931
Php: 8.0.29, pgsql: 14.0 (Debian 14.0-1.pgdg110+1), OS: Linux 5.19.0-50-generic x86_64
PHPUnit 9.5.28 by Sebastian Bergmann and contributors.

F                                                                   1 / 1 (100%)

Time: 00:00.848, Memory: 382.00 MB

There was 1 failure:

1) core_privacy\privacy\provider_test::test_table_coverage
The following tables with user fields must be covered with metadata providers:
  - local_recompletion_ccert_is (userid)

privacy/tests/privacy/provider_test.php:328
lib/phpunit/classes/advanced_testcase.php:80

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.
```

After:
```
❯ ctrl test --filter 'provider_test::test_table_coverage'
Moodle 4.1.5+ (Build: 20230826), bababe5ab3dfac3528ab55728ca8ee9315a06931
Php: 8.0.29, pgsql: 14.0 (Debian 14.0-1.pgdg110+1), OS: Linux 5.19.0-50-generic x86_64
PHPUnit 9.5.28 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 00:00.858, Memory: 382.00 MB

OK (1 test, 0 assertions)
```

**Comment by keevan** [comment]
Interestingly, phpmd had been enabled previously with no seeming issue:
```
      - name: Run phpmd
        if: ${{ always() }}
        run: moodle-plugin-ci phpmd
```
There are a number of concerns raised which might require some deeper refactoring.

Would you prefer the CI for this PR turn it off for now, and separate issues raised for those phpmd violations?

**Comment by keevan** [comment]
Turns out I misread the errors and phpcpd is the one returning an error.

Similar to how it has been running, I've disabled it so it doesn't make the CI red - with the idea is if it runs, it should be fixed and kept in a good state.

There are also some small modifications I made as I worked the file, in particular I moved the handling to a function and aligned the last bit of code for both bulkresetcompletion.php and editcompletion.php, so it's easier to compare and refactor at a later point in time if we want to tackle the phpcpd errors.

I can remove all those changes if you think it's easier to maintain in its previous state.

File: locallib.php [hunk 1]
```diff
@@ -89,3 +89,23 @@ function local_recompletion_get_data(array $data) {
...
+        $params = ['userid' => $user, 'course' => $courseid];
+        $ccompletion = new \completion_completion($params);
+        if ($ccompletion->is_complete()) {
+            // If we already have a completion date, clear it first so that mark_complete works.
+            $ccompletion->timecompleted = null;
+        }
+        $ccompletion->mark_complete($timecompleted);
+    }
+
+    return $result;
```
Comment (danmarsden) [inline locallib.php]: return $result? - probably just remove the return it doesn't need to return anything - (and update the phpdoc @return?)

Comment (keevan) [reply locallib.php]: haha where did that come from :eyes: 

Thanks for spotting that.

**Review by danmarsden — APPROVED** [review]

**Pull request merged**